app/models/host/managed.rb

```diff
@@ -880,16 +880,23 @@ def build_status_label(options = {})
   # rebuilds orchestration configuration for a host
   # takes all the methods from Orchestration modules that are registered for configuration rebuild
   # returns  : Hash with 'true' if rebuild was a success for a given key (Example: {"TFTP" => true, "DNS" => false})
-  def recreate_config
+  def recreate_config(only = nil)
     result = {}
-    Nic::Managed.rebuild_methods.map do |method, pretty_name|
+    nic_rebuild_methods = Nic::Managed.rebuild_methods
+    host_rebuild_methods = self.class.rebuild_methods
+    if only.present?
+      nic_rebuild_methods = nic_rebuild_methods.select { |k,v| only.include?(k) || only.include?(v) }
+      host_rebuild_methods = host_rebuild_methods.select { |k,v| only.include?(k) || only.include?(v) }
```

Comment:
Can you move the `select` statement into class method of Orchestration object named `rebuild_methods_for` or similar? That's the place it should be defined, will save one extra line as well.